File: CodingNinjas/Reverseeachword.cpp
#include <bits/stdc++.h> 

#include <iostream>

#include <algorithm>

#include <cstring>

using namespace std;

 

string reverseEachWord(string input)

{

    // Write your code here

    string result;

    string word;

 

    int n = input.size();

 

    for(int i=0 ; i<n ; i++) {

 

        if(input[i] == ' ')

        {

            reverse(word.begin(), word.end());

            result += word+' ';

            word = "";

        }

        word += input[i];

        if(input[i+1] == '\0')

        {

            reverse(word.begin(), word.end());

            result += word;

            word = "";

        }

    }

 

    return result;

}

 

int main()

{

 

    string str;

    getline(cin, str);

    string ans = reverseEachWord(str);

    cout << ans << endl;

    ;

}

File: CodingNinjas/Reverseeachword.java
import java.util.Scanner;

 

class Solution {

 

    static String reverseStringWordWise(String input) {

        // Write your code here

        String[] arr = input.split(" ");

        StringBuffer str = new StringBuffer();

        for(int i = arr.length-1;i >= 0;i--){

            str.append(arr[i]);

            str.append(" ");

        }

        return str.toString();

    }

 

    public static void main(String args[]) {

        Scanner sc = new Scanner(System.in);

        String input = sc.nextLine();

        String ans = reverseStringWordWise(input);

        System.out.println(ans);

    }

}
